state/syncer/baseAccountsSyncer.go

```diff
@@ -72,6 +74,9 @@ func checkArgs(args ArgsNewBaseAccountsSyncer) error {
 	if check.IfNil(args.Cacher) {
 		return state.ErrNilCacher
 	}
+	if check.IfNil(args.SyncStatisticsHandler) {
```

Comment:
Let's rename this to UserAccountsSyncStatisticsHandler as it only applies to the user accounts trie